dvc/scm/git/tree.py

```diff
@@ -118,7 +118,7 @@ def git_object_by_path(self, path):
             tree = tree[i]
         return tree
 
-    def _walk(self, tree, topdown=True):
+    def _walk(self, tree, topdown=True, dvcignore=None):
```

Comment:
Why are we passing dvcignore though? Why not just make IgnoreTree filter out what it gets from tree.walk?

Reply:
@pared You can tell it to not enter the dirs that you don't want, see how os.walk allows that.

Reply:
For the record: this was simply an old patch that I was too quick to review, so we are on the same page and changes are coming soon.